pkgs/development/python-modules/ray/bin.nix

```
@@ -0,0 +1,171 @@
+{ aiohttp
+, aiohttp-cors
+, aiorwlock
+, aiosignal
+, attrs
+, autoPatchelfHook
+, buildBazelPackage
+, buildPythonPackage
+, fetchPypi
+, click
+, cloudpickle
+, colorama
+, colorful
+, cython
+, dm-tree
+, fastapi
+, filelock
+, frozenlist
+, fsspec
+, gpustat
+, grpc
+, grpcio
+, gym
+, jsonschema
+, lib
+, lz4
+, matplotlib
+, msgpack
+, numpy
+, opencensus
+, packaging
+, pandas
+, py-spy
+, prometheus-client
+, protobuf3_20
+, psutil
+, pyarrow
+, pydantic
+, python
+, pythonRelaxDepsHook
+, pyyaml
+, requests
+, scikitimage
+, scipy
+, setproctitle
+, smart-open
+, starlette
+, stdenv
+, tabulate
+, tensorboardx
+, uvicorn
+, virtualenv
+
+  # optional dependencies
+, withData ? true
+, withServe ? true
+, withTune ? true
+  # rllib is tune with extra dependencies
+, withRllib ? true
+  # air is tune + data + train + serve
+, withAir ? true
+}:
+
+let
+  pname = "ray";
+  version = "2.0.0";
+  pyShortVersion = "cp${builtins.replaceStrings ["."] [""] python.pythonVersion}";
+  unsupported = throw "Unsupported system";
+  binary-hash = (import ./binary-hashes.nix)."${pyShortVersion}" or unsupported;
+
+  data-deps = [
+    pandas
+    pyarrow
+    fsspec
+  ];
+
+  serve-deps = [
+    aiorwlock
+    fastapi
+    pandas
+    starlette
+    uvicorn
+  ];
+
+  tune-deps = [
+    tabulate
+    tensorboardx
+  ];
+
+  rllib-deps = tune-deps ++ [
+    dm-tree
+    gym
+    lz4
+    matplotlib
+    scikitimage
+    pyyaml
+    scipy
+  ];
+
+  air-deps = data-deps ++ serve-deps ++ tune-deps ++ rllib-deps;
+in
+buildPythonPackage rec {
+  inherit pname version;
+  format = "wheel";
+  src = fetchPypi ({
+    inherit pname version format;
+    dist = pyShortVersion;
+    python = pyShortVersion;
+    abi = pyShortVersion;
+    platform = "manylinux2014_x86_64";
+  } // binary-hash);
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    pythonRelaxDepsHook
+  ];
+
+  pythonRelaxDeps = [ "grpcio" "click" "protobuf" ];
+
+  propagatedBuildInputs = [
+    attrs
+    aiohttp
+    aiohttp-cors
+    aiosignal
+    click
+    cloudpickle
+    colorama
+    colorful
+    cython
+    filelock
+    frozenlist
+    gpustat
+    grpcio
+    jsonschema
+    msgpack
+    numpy
+    opencensus
+    packaging
+    py-spy
+    prometheus-client
+    protobuf3_20
+    psutil
+    pydantic
+    pyyaml
+    requests
+    setproctitle
+    smart-open
+    virtualenv
+  ] ++ lib.optionals withData data-deps
+  ++ lib.optionals withServe serve-deps
+  ++ lib.optionals withTune tune-deps
+  ++ lib.optionals withRllib rllib-deps
+  ++ lib.optionals withAir air-deps;
+
+  postInstall = ''
+    prefix="$out/lib/${python.libPrefix}"/site-packages/ray/core/src/ray
+    chmod +x $prefix/gcs/gcs_server
+    chmod +x $prefix/raylet/raylet
+    chmod +x $prefix/thirdparty/redis/src/redis-server
```

```suggestion
    prefix=
    chmod +x $out/${python.sitePackages}/{ray/core/src/ray/gcs/gcs_server,raylet/raylet,thirdparty/redis/src/redis-server}
```
Why is there a redis server bundled into this package? This should be replaced with the nix version.